### PR TITLE
ER-250 Delete Draft Vacancy

### PR DESCRIPTION
**Pull request by chrisjensenuk**
- Added handler to manage the deletion of draft vacancies
- Added new 'are you sure?' page which is displayed when clicking the 'delete' link

### CODE REVIEW

File: src/Recruit.Vacancies.Client/Domain/Entities/Vacancy.cs [hunk 1]
```diff
@@ -17,6 +17,17 @@ public class Vacancy
...
         public DateTime? SubmittedDate { get; set; }
 
-        public bool CanSubmit => Status == VacancyStatus.Draft;
+        public bool IsDeleted { get; set; }
+        public DateTime? DeletedDate { get; set; }
+
+        /// <summary>
+        /// We can only submit draft vacancies that have not been deleted
+        /// </summary>
+        public bool CanSubmit => Status == VacancyStatus.Draft && IsDeleted == false;
```
Comment (sachpatel) [inline src/Recruit.Vacancies.Client/Domain/Entities/Vacancy.cs]: Do these field get persisted to the Vacancy in Mongo/Cosmos as well?

File: src/Recruit.Vacancies.Client/Domain/Entities/Vacancy.cs [hunk 2]
```diff
@@ -17,6 +17,17 @@ public class Vacancy
 
         public DateTime? SubmittedDate { get; set; }
 
-        public bool CanSubmit => Status == VacancyStatus.Draft;
+        public bool IsDeleted { get; set; }
```
Comment (sachpatel) [inline src/Recruit.Vacancies.Client/Domain/Entities/Vacancy.cs]: Shouldn't IsDeleted be derived from Status?

File: src/Employer.Web/Views/DeleteVacancy/Delete.cshtml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        <h1 class="heading-large">@Model.Title</h1>
+        <h1 class="heading-large">Delete vacancy</h1>
+        
+        <form asp-route="@RouteNames.DeleteVacancy_Delete_Post">
+            <div class="form-group">
+                <fieldset>
+                    <legend class="heading-medium">Are you sure?</legend>
+
+                    <div class="multiple-choice">
+                        <input asp-for="@Model.ConfirmDeletion" type="radio" value="true" id="delete-yes" />
```
Comment (sachpatel) [inline src/Employer.Web/Views/DeleteVacancy/Delete.cshtml]: Do we need to create our own `id` attribute, is it for styling? I thought the `asp-for` would generate it for us.

File: src/Recruit.Vacancies.Client/Domain/Entities/Vacancy.cs (hunk 1 above)
Comment (chrisjensenuk) [reply src/Recruit.Vacancies.Client/Domain/Entities/Vacancy.cs]: Nope. Just checked

File: src/Recruit.Vacancies.Client/Domain/Entities/Vacancy.cs (hunk 2 above)
Comment (chrisjensenuk) [reply src/Recruit.Vacancies.Client/Domain/Entities/Vacancy.cs]: Had a chat to Mark about this this morning. We want to know what status the vacancy was in before it was deleted so using a separate bool field for deletions

**Pull request merged**